src/Connection/Internal/Http2ConnectionProcessor.php

```diff
@@ -1326,12 +1326,6 @@ private function releaseStream(int $streamId, ?\Throwable $exception = null): vo
 
         unset($this->streams[$streamId]);
 
-        if ($streamId & 1) { // Client-initiated stream.
-            $this->remainingStreams++;
-
-            \assert($this->remainingStreams <= $this->concurrentStreamLimit);
-        }
-
```

Comment:
Why did you remove these lines?

Reply:
Because the streams $releaseCallback is set to Http2Processor::unreserveStream (Http2Connection line 60) which also increments $remainingStreams, subsequently the assertion would be false because $remainingStreams was incremented twice but only decremented once.

Either reserveStream and unreserveStream should *only* influence $reservedStreams, or they should be the only ones to influence both. I choose the later because I couldn't find a good place to decrement $remainingStreams elsewhere in Http2ConnectionProcessor

Reply:
@kelunik Checking in, does the above seem reasonable?

Reply:
I thought this was strange when I first looked at the PR, but yes, this seems correct if instead the unreserveStream callback is invoked upon stream release.